### DISCUSSION TITLE
How automatic PTR works

### DISCUSSION THREAD

**Discussion by kemeris2000** [sys4/netbox-plugin-dns#32] [Q&A] [answered]
My plugin instance does not generate PTR records, regardless of "Disable PTR" setting for A and AAAA records. Do I need to enable this functionality in Netbox configuration.py?

**Comment by peteeckel** [comment, marked as answer]
The PTR records will only be generated if the matching reverse zones are present - did you create the `in-addr.arpa` zone for the PTR?

**Comment by kemeris2000** [comment]
That was a case, I was expecting zone will be create automatically from @ A record. Thank you.

**Reply by peteeckel** [reply]
The problem with generating reverse zones automatically is that you can't know what zone to create from the IP address. 

Looking at the address `10.0.0.1`, you just don't know whether to create `10.in-addr.arpa`, `0.10.in-addr.arpa`, `0.0.10.in-addr.arpa` or even an RFC2317 zone (currently not supported by NetBox DNS, before you ask :-)). With IPv6 it's even less obvious. And in many cases (provider-managed reverse zones) it isn't such a good idea to create a local reverse zone.

So instead of trying the impossible and potentially failing and causing hard-to-track issues, NetBox DNS does not create reverse zones automatically at all.
